plugins/duo_admin/komand_duo_admin/util/api.py

```diff
@@ -0,0 +1,210 @@
+import requests
+import base64
+import hmac
+import hashlib
+import json
+import time
+from insightconnect_plugin_runtime.exceptions import PluginException
+from logging import Logger
+from urllib.parse import quote
+from datetime import datetime, timezone
+from komand_duo_admin.util.constants import Cause, Assistance
+from komand_duo_admin.util.exceptions import ApiException
+from komand_duo_admin.util.endpoints import (
+    ADMINISTRATOR_LOGS_ENDPOINT,
+    AUTHENTICATION_LOGS_ENDPOINT,
+    ENROLL_USER_ENDPOINT,
+    TRUST_MONITOR_EVENTS_ENDPOINT,
+    USER_ENDPOINT,
+    USER_PHONES_ENDPOINT,
+    USERS_ENDPOINT,
+)
+
+
+def rate_limiting(max_tries: int):
+    def _decorate(func):
+        def _wrapper(*args, **kwargs):
+            self = args[0]
+            retry = True
+            counter, delay = 0, 0
+            while retry and counter < max_tries:
+                if counter:
+                    time.sleep(delay)
+                try:
+                    retry = False
+                    return func(*args, **kwargs)
+                except ApiException as error:
+                    counter += 1
+                    delay = 2 ** (counter * 0.6)
+                    if error.cause == PluginException.causes[PluginException.Preset.RATE_LIMIT]:
+                        self.logger.info(f"Rate limiting error occurred. Retrying in {delay:.1f} seconds.")
+                        retry = True
+            return func(*args, **kwargs)
+
+        return _wrapper
+
+    return _decorate
```

Comment:
We can't because the function from SDK expects a PluginException to be raised when the request limit is exceeded, and this plugin raises a custom ApiException, so we need to use this function to handle it properly.